src/git.md

```diff
@@ -73,6 +73,30 @@ the same, with some steps skipped:
  2. Make, stage, and commit your additional changes just like before.
  3. Push those changes to your fork: `git push`.
 
+### Quick note about submodules
+
+When updating your local repository with `git pull`, you may notice that sometimes
+Git says you have modified some files that you have never edited. For example,
+running `git status` gives you something like (note the `new commits` mention):
+
+```
+On branch master
+Your branch is up to date with 'origin/master'.
+
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git restore <file>..." to discard changes in working directory)
+	modified:   src/tools/cargo (new commits)
+	modified:   src/tools/rls (new commits)
+	modified:   src/tools/rustfmt (new commits)
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+
+These changes are not changes to files: they are changes to submodules
+(more on this later). To get rid of those, run `git submodule update` (or run any
+`x.py` command which will automatically update the submodules).
+
```

Comment:
Github suggestions don't like nested triple quotes, but can you add this?

Note that there is currently a bug if you use `x.py`, worktrees, and submodules. If you run into an error like
```
error: failed to read `/home/joshua/rustc-worktree/src/tools/miri/cargo-miri/Cargo.toml`

Caused by:
  No such file or directory (os error 2)
```
it's not anything you did wrong. There is a workaround at [#77620](https://github.com/rust-lang/rust/issues/77620#issuecomment-705228229).

Reply:
@jyn514 Something I learned: you _can_ nest triple backticks if you use triple tildes as the outer delimiter:

~~~suggestion
```
foo
```
~~~

^ just for demonstration, don't apply!

Reply:
Works great, thanks!

Reply:
:o That's super helful, thank you ❤️